From ad48221d4e4aad1ff1fcaf356fb3407caf85ce80 Mon Sep 17 00:00:00 2001
From: Marvin Drees <marvin.drees@9elements.com>
Date: Thu, 5 Aug 2021 15:32:02 +0200
Subject: [PATCH] tmp421: Register sensor with thermal system

hwmon expects the first info channel to be a chip configuration for it
to consider it for the registration with the thermal system. Add that
channel configuration and set the thermal registration bit.

- Patch is property of Christian Svensson
---
 drivers/hwmon/tmp421.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/drivers/hwmon/tmp421.c b/drivers/hwmon/tmp421.c
index ede66ea6a730..06b7cb56dfb9 100644
--- a/drivers/hwmon/tmp421.c
+++ b/drivers/hwmon/tmp421.c
@@ -89,9 +89,11 @@ MODULE_DEVICE_TABLE(of, tmp421_of_match);
 struct tmp421_data {
 	struct i2c_client *client;
 	struct mutex update_lock;
+	u32 chip_config[2];
 	u32 temp_config[5];
+	struct hwmon_channel_info chip_info;
 	struct hwmon_channel_info temp_info;
-	const struct hwmon_channel_info *info[2];
+	const struct hwmon_channel_info *info[3];
 	struct hwmon_chip_info chip;
 	char valid;
 	unsigned long last_updated;
@@ -302,13 +304,19 @@ static int tmp421_probe(struct i2c_client *client)
 	if (err)
 		return err;
 
+	data->chip_config[0] = HWMON_C_REGISTER_TZ;
+
 	for (i = 0; i < data->channels; i++)
 		data->temp_config[i] = HWMON_T_INPUT | HWMON_T_FAULT;
 
 	data->chip.ops = &tmp421_ops;
 	data->chip.info = data->info;
 
-	data->info[0] = &data->temp_info;
+	data->info[0] = &data->chip_info;
+	data->info[1] = &data->temp_info;
+
+	data->temp_info.type = hwmon_chip;
+	data->temp_info.config = data->chip_config;
 
 	data->temp_info.type = hwmon_temp;
 	data->temp_info.config = data->temp_config;
-- 
2.32.0

